test/e2e/aws_test.go

```diff
@@ -132,13 +139,113 @@ var _ = Describe("functional tests", func() {
 			By("Scale the MachineDeployment up again")
 			scaleMachineDeployment(namespace, machineDeploymentName, scaleDownReplicas, initialReplicas)
 
+			By("Creating the LB service")
+			sess := getSession()
+			elbClient := elb.New(sess)
+			createLBService(namespace, clusterName, lbServiceName)
+			// This sleep is required to let the LB service up and running
+			time.Sleep(15 * time.Second)
+			verifyElbExists(elbClient, lbServiceName, true)
+
 			By("Deleting the Cluster")
 			deleteCluster(namespace, clusterName)
 
+			By("Checking provisioned LB deleted")
+			verifyElbExists(elbClient, lbServiceName, false)
 		})
 	})
 })
 
+func createLBService(namespace, clusterName, svcName string) {
+	cluster := &clusterv1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      clusterName,
+		},
+	}
+	kubeConfigData, err := kubeconfig.FromSecret(kindClient, cluster)
+	Expect(err).NotTo(HaveOccurred())
+
+	restConfig, err := clientcmd.RESTConfigFromKubeConfig(kubeConfigData)
+	Expect(err).NotTo(HaveOccurred())
```

Comment:
using the `kubeConfigPath` here will remove the effort of getting the secret again

Reply:
Added a new method which returns both configPath and cluster k8s client which were used across.